addon/-private/engine-ext.js

```diff
@@ -10,6 +11,7 @@ Engine.reopen({
     if (!(this instanceof Application)) {
       registry.register('component:link-to', EngineScopedLinkComponent);
       registry.register('component:link-to-external', ExternalLinkComponent);
+      registry.register('service:router', EngineRouterService);
```

Comment:
We should add an assertion / warning for apps that already inject a `router` service.

Reply:
I don't think so, with this landing new docs coming up to make explicit that every engine has a properly `router` injected.

Reply:
Exactly. You're not supposed to override framework services. However prior to this PR you could just inject / register your own `router` service. When these pre-existing apps upgrade, they should get an obvious error / warning, so that they can fix it.

Reply:
I think it's not our fault, the core team never recommended the users to use `router` service. Then I'd like to skip it for while and release it as soon as possible. Since we have this big lack around the engines ecosystem.